### ISSUE TITLE
View not destroyed when closing browser tab on page backed by omnifaces @ViewScoped

### ISSUE THREAD

**Issue by wsaxton** [omnifaces/omnifaces#208]
I set the view/logical view limits to 3 and deployed a simple JSF page with omnifaces ViewScoped backing bean.  Closing tabs does not appear to do anything.  I would have expected the secondary beans to have been destroyed so that the first bean is never automatically expired.

I'm using GF 4.1, JSF 2.2, Omnifaces 2.2

Steps to reproduce:

1. Open viewBean.xhtml in one tab
2. Open viewBean.xhtml in another tab, close the tab
3. Open viewBean.xhtml in another tab, close the tab
4. Open viewBean.xhtml in another tab, close the tab
5. Open viewBean.xhtml in another tab, close the tab
6. On the original tab, click the button.  Get the following error:

**javax.faces.application.ViewExpiredException: viewId:/viewBean.jsf - View /viewBean.jsf could not be restored.**

**web.xml:**
```
    <context-param>
        <param-name>com.sun.faces.numberOfViewsInSession</param-name>
        <param-value>3</param-value>
    </context-param>
    
    <context-param>
        <param-name>com.sun.faces.numberOfLogicalViews</param-name>
        <param-value>3</param-value>
    </context-param>
```
**ViewBean.java:**

```
package com.example.zunload;

import java.io.Serializable;
import javax.inject.Named;
import org.omnifaces.cdi.ViewScoped;

@Named
@ViewScoped
public class ViewBean implements Serializable {
    
    String testString = "Hello World";
    
    public ViewBean() {
    }
    
    public String getTestString() {
        return testString;
    }
    
    public void listener() {
        System.out.println("Do this");
    }
}
```

**viewBean.xhtml:**

```
<?xml version="1.0" encoding="UTF-8"?>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:h="http://xmlns.jcp.org/jsf/html">
    <h:head>
        <title>TODO supply a title</title>
        <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
    </h:head>
    <h:body>
        <div>#{viewBean.testString}</div>
        
        <h:form>
            <h:commandButton value="Submit" actionListener="#{viewBean.listener}"/>
        </h:form>
    </h:body>
</html>
```

**Comment by BalusC** [comment]
Reproduced. Nice find. Unfortunately, neither Mojarra nor MyFaces keeps server side view state "in sync" with PreDestroyViewMapEvent, so those server side view states still stick around. Your best bet is using client side state saving instead.

Theoretically, we could solve this by providing our own server side state manager, but this is not a trivial task. I'd rather first look how the JSF API itself can be improved with regard to this.

**Comment by BalusC** [comment]
For now, I added a hack to explicitly remove Mojarra's server side view state. Your use case now works fine for me. It's available in today's 2.3 snapshot. 

Will look at MyFaces later.

**Comment by wsaxton** [comment]
Thank you, @BalusC   I assume no release date for 2.3 as it will be released "when its good and ready" !

**Comment by BalusC** [comment]
When I'm 100% satisfied with o:socket. It's getting close :)

**Comment by BalusC** [comment]
MyFaces way of keeping track of views is kind of convoluted (see last commit), but your use case now also works fine for me in MyFaces.

**Comment by BalusC** [comment]
It's available in today's 2.3 snapshot.
